src/frontends/onnx/frontend/src/op/trilu.cpp

```diff
@@ -0,0 +1,104 @@
+// Copyright (C) 2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "op/trilu.hpp"
+
+#include "default_opset.hpp"
+#include "exceptions.hpp"
+#include "onnx_import/core/null_node.hpp"
+
+namespace ngraph {
+namespace onnx_import {
+namespace op {
+namespace set_1 {
+
+OutputVector trilu(const Node& node) {
+    const auto inputs = node.get_ng_inputs();
+    const auto num_inputs = inputs.size();
+
+    CHECK_VALID_NODE(node, num_inputs > 0 && num_inputs <= 2, "Trilu expects <= 2 input tensors. Got: ", num_inputs);
+
+    const auto& input = inputs[0];
+    const auto& rank = input.get_partial_shape().rank();
+    if (rank.is_static()) {
+        CHECK_VALID_NODE(node, rank.get_length() >= 2, "Trilu first input's rank must be >= 2");
+    }
+    bool is_k_available = num_inputs == 2 && !ngraph::op::is_null(inputs[1]);
+    if (is_k_available) {
+        CHECK_VALID_NODE(node, inputs[1].get_partial_shape().compatible({}), "Trilu second input must be a scalar");
+    }
+
+    const auto shape = std::make_shared<default_opset::ShapeOf>(input);
```

Comment:
I would suggest to explicitly state output type of ShapeOf as
1. k input of Trilu is int64
2. we do the same for Range operator since both Range and ShapeOf has additional input for output_type

`const auto shape = std::make_shared<default_opset::ShapeOf>(input, element::i64)`